lmdb: Invalid reuse of reader locktable slot

```
    @contextmanager
    def describe(self,
                 record_type: Type['DatastoreRecord'],
                 record_id: Union[int, str],
                 writeable: bool = False) -> Type['DatastoreRecord']:
        """
        This method is used to perform CRUD operations on the datastore within
        the safety of a context manager by returning an instance of the
        `record_type` identified by the `record_id` provided.
    
        When `writeable` is `False`, the record returned by this method
        cannot be used for any operations that write to the datastore. If an
        attempt is made to retrieve a non-existent record whilst `writeable`
        is `False`, this method raises a `RecordNotFound` error.
    
        When `writeable` is `True`, the record can be used to perform writes
        on the datastore. In the event an error occurs during the write, the
        transaction will be aborted and no data will be written, and a
        `DatastoreTransactionError` will be raised.
    
        If the record is used outside the scope of the context manager, any
        writes or reads will error.
        """
        with suppress(ValueError):
            # If the ID can be converted to an int, we do it.
            record_id = int(record_id)
    
>       with self.__db_env.begin(write=writeable) as datastore_tx:
E       lmdb.BadRslotError: mdb_txn_begin: MDB_BAD_RSLOT: Invalid reuse of reader locktable slot

nucypher/datastore/datastore.py:125: BadRslotError
```

See full traceback in CI Build: 

https://app.circleci.com/pipelines/github/nucypher/nucypher/5509/workflows/1994c9bf-611f-4e3b-8732-39ecc67418c0/jobs/97113

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -76,12 +76,12 @@ class TestLMDBEnv:
     @contextmanager
     def begin(self, *args, **kwargs):
         try:
-            self.thread_lock.acquire()
-            with self.open() as lmdb_env:
-                with lmdb_env.begin(*args, **kwargs) as lmdb_tx:
-                    yield lmdb_tx
+            with self.thread_lock:
+                with self.open() as lmdb_env:
+                    with lmdb_env.begin(*args, **kwargs) as lmdb_tx:
+                        yield lmdb_tx
         finally:
-            self.thread_lock.release()
+            pass
 
     def path(self):
         return self.db_path

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -67,16 +67,16 @@ class TestLMDBEnv:
     __test__ = False    # Prohibit pytest from collecting this
 
     LMDB_OPEN_FUNC = lmdb.open
+    THREAD_LOCK = threading.Lock()
 
     def __init__(self, *args, **kwargs):
         self.db_path = args[0]
         self.open = partial(self.LMDB_OPEN_FUNC, *args, **kwargs)
-        self.thread_lock = threading.Lock()
 
     @contextmanager
     def begin(self, *args, **kwargs):
         try:
-            with self.thread_lock:
+            with self.THREAD_LOCK:
                 with self.open() as lmdb_env:
                     with lmdb_env.begin(*args, **kwargs) as lmdb_tx:
                         yield lmdb_tx

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -18,6 +18,7 @@ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 import lmdb
 import os
 import pytest
+import threading
 from contextlib import contextmanager
 from eth_account.account import Account
 from functools import partial
@@ -70,15 +71,17 @@ class TestLMDBEnv:
     def __init__(self, *args, **kwargs):
         self.db_path = args[0]
         self.open = partial(self.LMDB_OPEN_FUNC, *args, **kwargs)
+        self.thread_lock = threading.Lock()
 
     @contextmanager
     def begin(self, *args, **kwargs):
         try:
+            self.thread_lock.acquire()
             with self.open() as lmdb_env:
                 with lmdb_env.begin(*args, **kwargs) as lmdb_tx:
                     yield lmdb_tx
         finally:
-            pass
+            self.thread_lock.release()
 
     def path(self):
         return self.db_path

